src/reducers/articles.reducer.js

```diff
@@ -1,39 +1,73 @@
 import { combineReducers } from 'redux';
-import state from './initialState';
+import initialState from './initialState';
 import constants from '../constants';
 import createPromiseReducer from '../utils/createPromiseReducer.util';
+import uniqueBySlug from '../utils/uniqueBySlug.util';
+import nRandom from '../utils/nRandom.util';
 
 const {
   POST_ARTICLE,
   FETCH_ARTICLE,
   UPDATE_ARTICLE,
   FETCH_ALL_ARTICLES,
 } = constants;
-const initialState = state.articles;
 
 const postArticle = createPromiseReducer({
-  initialState,
+  initialState: initialState.article,
   actionName: POST_ARTICLE,
   dataField: 'article',
 });
 
 const fetchArticle = createPromiseReducer({
-  initialState,
+  initialState: initialState.article,
   actionName: FETCH_ARTICLE,
   dataField: 'article',
 });
 
 const updateArticle = createPromiseReducer({
-  initialState,
+  initialState: initialState.article,
   actionName: UPDATE_ARTICLE,
   dataField: 'article',
 });
 
-const fetchAllArticles = createPromiseReducer({
-  initialState,
-  actionName: FETCH_ALL_ARTICLES,
-  dataField: 'article',
-});
+const { allArticles: allArticlesInitialState } = initialState;
+const fetchAllArticles = (state = allArticlesInitialState, action) => {
+  const { type, payload } = action;
+  switch (type) {
+    case `${FETCH_ALL_ARTICLES}_PENDING`:
+      return {
+        ...state,
+        loading: true,
+      };
+    case `${FETCH_ALL_ARTICLES}_FULFILLED`:
+      return {
+        ...state,
+        loading: false,
+        success: true,
+        articles: {
+          ...payload.data.article,
+          results: [...payload.data.article.results].sort(() => (0.5 - Math.random())),
```

Comment:
Isn't this the same function as `nRandom.util.js`?

Reply:
Not quite, the `nRandom` randomly picks `n` articles from an array, this one just shuffles the array.

Reply:
What does this function do?

Also, let's have 0.5 as a constant

Reply:
The function, it just shuffles the elements of an array. 

When sorting with `Array.prototype.sort`, one might choose to implement a function that receives two elements of the array and returns a negative number if the first element is smaller than the second, positive if the second is larger and 0 if equal.

Since `Math.random` returns a number `R` in the range [0, 1), we rely on that value to randomly sort the elements of the array by doing `0.5 - R` to randomly return a negative/positive value which shuffles the elements of the array.